pkg/Makefile

```diff
@@ -3,11 +3,17 @@ DIRS = $(dir $(shell find . -maxdepth 2 -mindepth 2 -type f -name build.yml))
 
 OPTIONS ?=
 
+PUSHOPTIONS =
```

Comment:
This is not necessary since we already have OPTIONS, which was added for this purpose and is more generic. See commit
```
1fadf13ec4a2 ("pkg: Allow specifying addition options to 'linuxkit' invocations")
```

Reply:
I tried every which way, but make kept stymieing me. How would you do it such that it would work with all 4 combinations of: `LK_RELEASE` set and not set; `OPTIONS` set and not set?

If you do:

```make
OPTIONS += -release $(LK_RELEASE)
```

then it only works if `OPTIONS` was not set, i.e. `make LK_RELEASE=foo` but not `make LK_RELEASE=foo OPTIONS=abc`.

If you do

```make
OPTIONS = $(OPTIONS) -release $(LK_RELEASE)
```

you end up with a recursive issue, etc.

If you can find a way to do it such that it works in all circumstances, I am happy to use it. But for now, this does just work.

Reply:
if you need to specify more than one option (where do you need this?) can'y you do: `make OPTIONS="--foo -release $(LK_RELEASE)"`?

Reply:
I actually do not know that we do. I am trying to keep the current behaviour while adding capabilities.

I also don't want to have to tell people `make push`, unless you are doing a release, in which case do `make OPTIONS="--release $(LK_RELASE)" push`. This alpine and its dependency updating and pushing process is so brutal and complicated that anything that makes it easier is of value.

If I can tell people (which I did in the doc): follow the steps in here whatever you are doing. If you are creating a release first, `export LK_RELEASE=foo` and then follow the _exact same steps_. I am 200% for it.

Reply:
we do push without release. any package update should be a push (and update of tags) and release should have once in a while once  we have a good set up updated packages.

The alpine part is really a separate issue which has little to do with the Makefile changes.

Reply:
> we do push without release. 

Not according to the existing docs. They just said "run `make OPTIONS="-release ${LK_RELEASE" push`. So I could not even copy-paste, as confusing as that would be, because I would create a release by accident, so I had to carefully copy, edit, paste.

In documenting what is needed, it was clear to me that there are two use cases:

* push when not releasing `make push`
* push when releasing `make OPTIONS="-release rel_${LK_RELEASE}" push`

Trying to simplify this into something that requires zero cognitive energy is exactly what I am after. Always run `make push`, but sometimes you set `LK_RELEASE` at the very beginning of the process.

Reply:
the doc is about releasing but in between releases we update packages and then update the YAMLs with the new tags, so a release is really something we do two times a year or so while some packages do get updated much more frequent.

Reply:
Which is why I broke it out into a separate doc for alpine, and tried to simplify just two of the commands. I am all up for a script for all of it, but can we do this one step at a time? I most certainly am not getting to scripting it all up for a long time. I would rather capture some simplifications, however small, right now.

Which reminds me; I need to rename the file like you suggested.